### PR TITLE
Documented test result metadata in UI

### PR DESCRIPTION
**Pull request by validbeck**
## Internal Notes for Reviewers

> For sc-4768, I:
> - Created a net-new page discussing working with test results (& viewing their metadata): `developer/model-documentation/work-with-test-results.qmd`
> - Retooled `developer/model-documentation/generate-model-documentation.qmd` to have a cleaned up workflow only (with much less confusing instructions!)
> - Then, created a new landing page (`developer/model-documentation/generating-model-documentation.qmd`) for the conceptual workflow overview and shoved all the relevant articles under here

## Work with test results
- This page expands on some of the content from "Work with content blocks" — which I cleaned up a little to link out to this guide for test-driven blocks instead of having a section of confusing conditional instructions.
- After laying the foundation I then expanded on how to view the test result metadata

| Add test results | View test result metadata |
|---|---|
| Screenshot 2024-08-22 at 10 01 36 AM | Screenshot 2024-08-22 at 9 53 11 AM|

### Work with content blocks
Since there's now an extended guide on test-driven blocks, I simplified this general context blocks guide to make room for a later potential overview of the text editor.

| Old | New |
|---|---|
| Screenshot 2024-08-22 at 10 03 19 AM | Screenshot 2024-08-22 at 9 53 24 AM|

## Generate model documentation
- The old version of this page was a bit clunky, and the instructions jumped back and forth and were a bit hard to follow. I broke down the workflow into the overview and the actual instructions, then cleaned up the instructions to be hopefully a bit more parsable.

| Old | New |
|---|---|
| Screenshot 2024-08-22 at 10 13 58 AM | Screenshot 2024-08-22 at 9 52 23 AM|

## Generating model documentation
- This is the new conceptual overview page, cleaned up, serving as the landing for all the articles relevant to the documentation generation workflow. 
- The initial version of this guide was named "Model documentation workflow" but I think the action gerund is a bit clearer.

| Old | New |
|---|---|
| Screenshot 2024-08-22 at 10 14 41 AM | Screenshot 2024-08-22 at 10 27 08 AM |
| Screenshot 2024-08-22 at 10 14 50 AM | Screenshot 2024-08-22 at 9 51 33 AM |

### Install and initialize the developer framework > Install and initialize ValidMind
- I noticed that when wrapped in a breadcrumb the variable is bœrked... not great, and while playing around with just plain text I think just using the simple product name is snappier, and sort of hammers home the "cohesion" element of the framework / UI being two pieces of the same puzzle.

| Old | New |
|---|---|
| Screenshot 2024-08-22 at 9 40 29 AM | Screenshot 2024-08-22 at 9 52 32 AM |

### Get started 
- Fixed up the `developer/get-started-developer-framework.qmd` to have the new "Generating model documentation" at the bottom and made the listings look a bit more uniform

| Old | New |
|---|---|
| Screenshot 2024-08-22 at 10 11 31 AM | Screenshot 2024-08-22 at 9 37 33 AM|

### Navigation drop-down
- Added the new "Generating model documentation" landing to the developer framework drop-down

| Old | New |
|---|---|
| Screenshot 2024-08-22 at 10 10 40 AM | Screenshot 2024-08-22 at 10 08 36 AM |

### Prerequisties
- I cleaned up the prereq sections for all the guides under "Generating model docuemntation"
> Please note I noticed the extra comma while I was capping for this section, so if you see it in any other screenshots it's been resolved!

| Old | New |
|---|---|
| Screenshot 2024-08-22 at 10 18 32 AM | Screenshot 2024-08-22 at 10 22 47 AM|
| Screenshot 2024-08-22 at 10 23 07 AM | Screenshot 2024-08-22 at 10 23 02 AM |

### Jupyter notebooks > Jupyter Notebooks
- I thought I had already changed this, but the actual proper noun (according to JH website) is "Jupyter Notebooks" so I've just adjusted our verbiage to match that.

### CODE REVIEW

**Comment by validbeck** [comment]
Hi @even-steven! Just wanted to check in with you that these instructions are accurate: 

| Add test results | View test result metadata |
|---|---|
| Screenshot 2024-08-22 at 10 01 36 AM | Screenshot 2024-08-22 at 9 53 11 AM|

**Comment by even-steven** [comment]
> Hi @even-steven! Just wanted to check in with you that these instructions are accurate:
> 
> Add test results	View test result metadata
> Screenshot 2024-08-22 at 10 01 36 AM	Screenshot 2024-08-22 at 9 53 11 AM

Hey @validbeck, looks great! One thing that might be worth mentioning is that `Validators` in the model can add test-driven blocks within `Validation Reports` and `Developers` can add test-driven blocks within the `Documentation`. 

Very nice 👏🏽

**Comment by validbeck** [comment]
Thanks @even-steven, that's definitely info I didn't have! 

Screenshot 2024-08-22 at 1 25 51 PM

File: site/developer/model-documentation/generating-model-documentation.qmd [hunk 1]
```diff
@@ -0,0 +1,101 @@
+---
+title: "Generating model documentation"
```
Comment (nrichers) [inline site/developer/model-documentation/generating-model-documentation.qmd:2]:
```suggestion
title: "Documenting models"
```

This topic covers more than just _documentation generation_ which is just step 2. in what's being described. A more encompassing title would be the suggestion I made, but you will need to update the title and any related stuff like _quarto.yml and links/listings.

File: site/developer/model-documentation/generate-model-documentation.qmd [hunk 2]
```diff
@@ -1,128 +1,77 @@
 ---
 title: "Generate model documentation"
```
Comment (nrichers) [inline site/developer/model-documentation/generate-model-documentation.qmd:2]: This topic should also be renamed since it covers much more than just the _generation_ of documentation. This is effectively the more hands-on end-to-end workflow for documenting a model. 

```suggestion
title: "Document models"
```

File: site/developer/model-documentation/generate-model-documentation.qmd [hunk 3]
```diff
@@ -1,128 +1,77 @@
...
-         - For new block:
-            - Add new editable text content block
-            - Review and collaborate on the content block
-         - For existing blocks: Select from available texts from content provider
+         a. For new block:
+            1. Add new editable text content block
+            2. Review and collaborate on the content block
+         b. For existing blocks: Select from available texts from content provider
+
+2. [Submit your model documentation for review](/guide/model-documentation/submit-for-approval.qmd).
```
Comment (nrichers) [inline site/developer/model-documentation/generate-model-documentation.qmd:56]: This is under section 4. in the parent topic, so it should not be under the third section here.

File: site/developer/model-documentation/install-and-initialize-validmind.qmd [hunk 4]
```diff
@@ -1,17 +1,17 @@
 ---
-title: "Install and initialize the {{< var vm.developer >}}"
+title: "Install and initialize ValidMind"
```
Comment (nrichers) [inline site/developer/model-documentation/install-and-initialize-validmind.qmd:2]: This feels like it simultaneously overloads and underspecifies what the name "ValidMind" signifies. How about being very descriptive and simply using the term we already have — "client library"?

File: site/_quarto.yml [hunk 5]
```diff
@@ -42,6 +42,8 @@ website:
           file: developer/get-started-developer-framework.qmd
         - text: "{{< fa cubes >}} Supported Models"
           file: developer/model-documentation/supported-models.qmd
+        - text: "{{< fa book-medical >}} Generating Model Documentation"
```
Comment (nrichers) [inline site/_quarto.yml:45]: I don't recall why we ever used title-style caps for this dropdown menu (maybe because the navbar uses them? 🤔), but perhaps this would be a good time to switch to sentence-style caps since we're referencing actual topic titles?

File: site/developer/model-documentation/generate-model-documentation.qmd [hunk 6]
```diff
@@ -1,128 +1,77 @@
 ---
 title: "Generate model documentation"
 date: last-modified
+number-sections: true
```
Comment (nrichers) [inline site/developer/model-documentation/generate-model-documentation.qmd:4]: I don't think `number-sections: true` really works if numbering starts at `0.1`. I'd either number manually (my preference) or not bother.

image

**Review by nrichers — CHANGES_REQUESTED** [review]
This PR moves the needle on some protracted content structure issues we have in our developer doumentation. Very nice! 

I am going to request some changes, mainly to debunk some of our topic titles further and to sort out the section numbering (happy to chat more if you think it would be useful).

File: site/_quarto.yml (hunk 5 above)
Comment (validbeck) [reply site/_quarto.yml:45]: Yeah, I just followed what was already there — I kind of like it this way because like you say the navbar has them, and this makes it consistent, but up to you...

File: site/developer/model-documentation/generate-model-documentation.qmd (hunk 3 above)
Comment (validbeck) [reply site/developer/model-documentation/generate-model-documentation.qmd:56]: I'm a little confused, there isn't a section 4 in the original?

File: site/developer/model-documentation/generate-model-documentation.qmd (hunk 2 above)
Comment (validbeck) [reply site/developer/model-documentation/generate-model-documentation.qmd:2]: Should I change the filename as well?

**Comment by validbeck** [comment]
Changes:

Screenshot 2024-08-22 at 4 28 40 PM
Screenshot 2024-08-22 at 4 29 06 PM
Screenshot 2024-08-22 at 4 29 20 PM

**Review by nrichers — APPROVED** [review]
LGTM after your latest revisions. Thank you! 🚀
